src/app/query/bitcoin/ordinals/brc20/brc20-tokens.query.ts

```diff
@@ -53,15 +61,65 @@ async function fetchBrc20TokensByAddress(address: string): Promise<Brc20Token[]>
   });
 }
 
-type FetchBrc20TokensByAddressResp = Awaited<ReturnType<typeof fetchBrc20TokensByAddress>>;
+export function useBrc20TokensQuery() {
+  const network = useCurrentNetwork();
+  const nativeSegwitSigner = useCurrentAccountNativeSegwitIndexZeroSigner();
+  const currentBitcoinAddress = nativeSegwitSigner.address;
+  const createSigner = useCurrentAccountTaprootSigner();
+
+  const getTaprootAddressData = useCallback(
+    (fromIndex: number, toIndex: number) => {
+      return createNumArrayOfRange(fromIndex, toIndex - 1).map(num => {
+        // TO-DO remove this check when we have a better way to handle this
+        if (!createSigner) return '';
```

Comment:
If you `if (!createSigner) throw new Error('No signer');` outside this fn, do you still need the check?